front/src/applications/operationalStudies/hooks/useScenarioData.ts

```diff
@@ -160,13 +160,29 @@ const useScenarioData = (scenario: ScenarioResponse, infra: InfraWithState) => {
     [trainIdUsedForProjection, timetableItems]
   );
 
-  const trainScheduleEditoastIds = useMemo(
-    () => (fetchedTrainSchedulesResults || []).map((trainSchedule) => trainSchedule.id),
+  const formattedTrainScheduleIds = useMemo(
+    () =>
+      (fetchedTrainSchedulesResults || []).map((trainSchedule) =>
+        formatEditoastTrainIdToTrainScheduleId(trainSchedule.id)
+      ),
     [fetchedTrainSchedulesResults]
   );
 
+  const formattedPacedTrainIds = useMemo(
+    () =>
+      (fetchedPacedTrains || []).map((pacedTrain) =>
+        formatEditoastTrainIdToPacedTrainId(pacedTrain.id)
+      ),
+    [fetchedPacedTrains]
+  );
+
+  const timetableItemIds = useMemo(
+    () => [...formattedTrainScheduleIds, ...formattedPacedTrainIds],
+    [formattedTrainScheduleIds, formattedPacedTrainIds]
+  );
```

Comment:
Why not this ? :)
```suggestion
  const timetableItemIds = useMemo(
    () => timetableItems.map((item) => item.id),
    [timetableItems]
  );
```

Then you can drop `formattedTrainScheduleIds` and `formattedPacedTrainIds`

Reply:
Hmm, because I like to complicate my life ? 🙃 Done